### PR TITLE
Courses page

### PR DESCRIPTION
**Pull request by vunn-1010**
Course page show:
- courses that user has enrolled
- all courses in the system

[Ticket Redmine](https://edu-redmine.sun-asterisk.vn/issues/24564)

Layout view
image

### CODE REVIEW

**Comment by vunn-1010** [comment]
ready

File: app/Repositories/CourseRepository.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    /**
+     * Get all courses with enrollment status of given users
+     * 
+     * @param \App\Models\User $user
+     * @return \Illuminate\Database\Eloquent\Collection
+     */
+    public function getAll($user)
+    {
+        $courses = Course::simplePaginate(10);
```
Comment (quoctd-0607) [inline app/Repositories/CourseRepository.php:32]: config setting cho 10

**Comment by vunn-1010** [comment]
ready

File: routes/web.php [hunk 2]
```diff
@@ -28,3 +28,5 @@
 Route::prefix('wordlist')->group(function () {
     Route::get('/', 'WordListController@index')->name('wordlist.index');
 });
+
+Route::resource('courses', 'CourseController');
```
Comment (quoctd-0607) [inline routes/web.php:32]: dùng only với những action mình dùng. tránh thừa resources

**Comment by vunn-1010** [comment]
ready

**Pull request merged**